plugins/modules/network/icx/icx_firmware_upgrade.py

```diff
@@ -0,0 +1,243 @@
+#!/usr/bin/python
+# Copyright: Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+DOCUMENTATION = """
+---
+module: icx_firmware_upgrade
+version_added: "1.2.0"
+author: "Ruckus Wireless (@Commscope)"
+short_description: Upgrades firmware of ICX switches
+description:
+  - This module copies new firmware to switch, flashes it to primary/secondary and reboots the switch.
+notes:
+  - Tested against ICX 10.1.
+  - For information on using ICX platform, see L(the ICX OS Platform Options guide,../network/user_guide/platform_icx.html).
+options:
+  server_type:
+    description:
+      - Data transfer protocol to be used
+    type: str
+    choices: ['scp', 'https', 'tftp']
+    required: true
+  server_address:
+    description:
+      - IPV4/IPV6 address of the scp/https/tftp server
+    type: str
+    required: true
+  server_port:
+    description:
+      - The port number of the server. Default values will be selected based on protocol type.
+        Default scp:22, https:443
+    type: int
+  partition:
+    description:
+     - Partition to be used for upgrade.
+    type: str
+    choices: ['primary','secondary','fips-ufi-primary-sig','fips-ufi-secondary-sig']
+  scp_user:
+    description:
+      - remote username to be used for scp login.
+    type: str
+  scp_pass:
+    description:
+      - remote password to be used for scp login.
+    type: str
+  boot_only:
+    description:
+      - Boot the switch if True.
+    default: no
+    type: bool
+  save_running_config:
+    description:
+      - execute Write memory Command.
+    default: no
+    type: bool
+  filename:
+    description:
+      - The path of the .bin/.sig file.
+    type: str
+    required: true
+"""
+
+EXAMPLES = """
+- name: upgrade firmware using scp
+  community.network.icx_firmware_upgrade:
+    server_type: scp
+    server_address: 10.20.1.1
+    partition: secondary
+    filename: SPR08030.bin
+    boot_only: False
```

Comment:
```suggestion
    boot_only: false
```
Please use `true`/`false` in documentation.